internal/controllers/module_reconciler.go

```diff
@@ -75,17 +81,8 @@ func (mr *ModuleReconciler) Reconcile(ctx context.Context, mod *kmmv1beta1.Modul
 
 	logger.Info("Starting Module reconciliation")
 
-	if mod.GetDeletionTimestamp() != nil {
-		//Module is being deleted
-		if err := mr.nsLabeler.tryRemovingLabel(ctx, mod.Namespace, mod.Name); err != nil {
-			return ctrl.Result{}, fmt.Errorf("error while trying to remove the label on namespace %s: %v", mod.Namespace, err)
-		}
-
-		err := mr.reconHelper.finalizeModule(ctx, mod)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to finalize Module %s/%s: %v", mod.Namespace, mod.Name, err)
-		}
-		return ctrl.Result{}, nil
+	if err := mr.reconHelper.handleModuleInDeletionProcess(ctx, mod); err != nil {
```

Comment:
i think that handleModuleInDeletionProcess should not check if the mod is getting deleted. The "if should stay in Reconciler, and the handling of "if" should be done in handleModuleInDeletionProcess

Reply:
And if the method is renamed to `handleModuleInDeletionProcessIfNeeded`? The fact that the comment saying `the module is being deleted` is there is because the check of the timestamp isn't clear. Putting it in a method that its name describe the logic is a good wrapping IMO.